File: pytools/examples/csharp/client/Protocol/ReqChatSend.cs
using System.Collections;
using System.Collections.Generic;


public class ReqChatSend
{
	private byte _channel;
	private uint _dest_uid;
	private string _content;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteByte(this._channel);
		packet.WriteUint(this._dest_uid);
		packet.WriteString(this._content);
		packet.Encode(Msg.P_REQ_CHAT_SEND);
		return packet;
	}


	public byte channel
	{
		get { return this._channel; }
		set { this._channel = value; }
	}

	public uint dest_uid
	{
		get { return this._dest_uid; }
		set { this._dest_uid = value; }
	}

	public string content
	{
		get { return this._content; }
		set { this._content = value; }
	}

}

File: pytools/examples/java/client/protocol/ReqChatSend.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqChatSend
{
	private byte channel;
	private int dest_uid;
	private String content;


	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeByte(channel);
		packet.writeInt(dest_uid);
		packet.writeString(content);
		return packet.encode((short)Msg.P_REQ_CHAT_SEND);
	}



	public void setChannel(byte channel)
	{
		this.channel = channel;
	}

	public void setDestUid(int dest_uid)
	{
		this.dest_uid = dest_uid;
	}

	public void setContent(String content)
	{
		this.content = content;
	}

}
